lib/js/controllers/s-popover.js

```diff
@@ -0,0 +1,94 @@
+(function () {
+    "use strict";
+    Stacks.addController("s-popover", {
+        targets: [],
+        
+        /**
+         * Initializes popper.js and document events on controller connect
+         */
+        connect: function() {
+            var popoverId = this.element.getAttribute("aria-controls");
+
+            if (!popoverId) {
+                throw "[aria-controls=\"{POPOVER_ID}\"] required";
+            }
+
+            this.popoverElement = document.getElementById(popoverId);
+
+            this.popper = new Popper(this.element, this.popoverElement, {
+                placement: this.data.get("placement") || "bottom",
+            });
+
+            // in order for removeEventListener to remove the right event, this bound function needs a constant reference
+            this._boundClickFn = this._hideOnOutsideClick.bind(this);
+
+            document.addEventListener("click", this._boundClickFn);
```

Comment:
balpha had concerns about having event listeners bound to every popover on the page, even if it's not active.

See https://gh.stackoverflow.com/Core/StackOverflow/blob/bbb139746376b76a8ac531ce1b37434e5178b220/StackOverflow/Content/Js/PartialJS/full-common/controllers/se-popover.js#L145 for how I'm binding and unbinding events between connect and toggle.

Reply:
As an aside, have we considered touch events here?

Reply:
Ah, yes. Good catch. Touching outside on a mobile device doesn't work as expected.

Reply:
Touch events are tricky since they don't always mean what you think they will.  Most touch events are scrolling.  Similarly, if you make a change too early with a touch event, the click event will still fire in that exact same position, possibly on an entirely different element.

The specific problem we're experiencing here is that mobile browsers only fire `click` events if you touch an element with a `click` event, don't drag, then release focus.  I believe `click` also fires after `touchup`.  If you touch whitespace, there's no `click` event the event can't bubble up to `document`.  Touching a link or button on the other hand does work to hide the popover.

One solution _could_ be to move the dismiss event to the `body` element:

```javascript
document.querySelector('body').addEventListener("click", this._boundClickFn);
```

But I wouldn't do that. I would just do nothing. This kind of behavior is standard for mobile users and they are more than capable of clicking the button again.

Reply:
@bnickel brings up some excellent points on this. I'm punting on touch events for now because they're a pain. We can revisit later if we feel strongly about it.